async-nats/src/jetstream/context.rs

```diff
@@ -90,6 +90,44 @@ impl Context {
             Response::Ok(info) => Ok(Stream { info }),
         }
     }
+
+    pub async fn delete_stream<T: AsRef<str>>(
+        &mut self,
+        stream: T,
+    ) -> Result<DeleteResponse, Error> {
+        let stream = stream.as_ref();
+        if stream.is_empty() {
+            return Err(Box::new(io::Error::new(
+                ErrorKind::InvalidInput,
+                "the stream name must not be empty",
+            )));
+        }
+        let subject = format!("{}.STREAM.DELETE.{}", self.prefix, stream);
+        match self.request(subject, &json!({})).await? {
+            Response::Err { error } => Err(Box::new(std::io::Error::new(
+                ErrorKind::Other,
+                format!(
+                    "nats: error while deleting a stream: {}, {}",
+                    error.code, error.description
+                ),
+            ))),
+            Response::Ok(delete_response) => Ok(delete_response),
+        }
+    }
+
+    pub async fn update_stream(&mut self, config: &StreamConfig) -> Result<StreamInfo, Error> {
```

Comment:
Update should probably be able to take stream info (via traits)?

```suggestion
    pub async fn update_stream(&mut self, config: &StreamConfig) -> Result<StreamInfo, Error> {
```

Reply:
We could, but why we should?

StreamInfo have nested config anyway and user would always alter that part before passing it here.

They can call it with `update_stream(&info.config)` but still, I would create a new config each time I alter it, instead of altering info directly.